src/librustc/ty/instance.rs

```diff
@@ -47,7 +47,7 @@ pub enum InstanceDef<'tcx> {
     ///`<T as Clone>::clone` shim for arrays and tuples
     CloneStructuralShim(DefId, Ty<'tcx>),
     ///`<T as Clone>::clone` shim for closures
-    CloneNominalShim(DefId, Ty<'tcx>),
+    CloneNominalShim { clone: DefId, ty: DefId },
```

Comment:
comment about what `ty` is the def-id of...

Reply:
The nominal type definition (closure, ADT, etc. - @Manishearth has a branch where enums also want this). Could use a better field name.